datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -708,7 +708,7 @@ pub fn try_type_union_resolution_with_struct(
 /// strings.  For example when comparing `'2' > 1`,  the arguments will be
 /// coerced to `Utf8` for comparison
 pub fn comparison_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
-    if lhs_type == rhs_type {
+    if lhs_type.equals_datatype(rhs_type) {
```

Comment:
👍 

I also want to ping @jayzhan211 , He corrected me about a similar thing before.

Reply:
Since we only care about data type in this coercion, so I think this makes sense